Cargo.lock

```diff
@@ -2615,6 +2615,7 @@ dependencies = [
  "consumption_metrics",
  "crc32c",
  "criterion",
+ "crossbeam-channel",
```

Comment:
I don't have a problem with including crossbeam (it is a very reputable crate), but I think we could manage with a tokio channel if we wanted to.

The receiver try_recv requires &mut, but that can be addressed by spawning a task that owns the receiver, and has a reference to the page cache.  Then that task just sits in a receive loop, and calls into page cache to process the entries with `drop_buffers_for_immutable`